packages/service-worker/worker/test/happy_spec.ts

```diff
@@ -723,30 +684,242 @@ describe('Driver', () => {
+                    'foo': {'operation': 'lastFocusedOrOpen', 'url': '/foo'},
+                  },
+                },
+              },
+              'foo');
+          expect(scope.clients.openWindow).toHaveBeenCalledWith('http://localhost:4200/foo');
+        });
+        it('fallback to openWindow at `/` when no last client and no `url`', async () => {
+          expect(await makeRequest(scope, '/foo.txt')).toEqual('this is foo');
+          spyOn(scope.clients, 'openWindow');
+          spyOn(scope.clients, 'matchAll').and.returnValue(Promise.resolve([]));
+
+          await driver.initialized;
+          await scope.handleClick(
+              {
+                title: 'This is a test without action',
+                body: 'Test body without action',
+                'data': {
+                  'onActionClick': {
+                    'foo': {'operation': 'lastFocusedOrOpen'},
+                  },
+                },
+              },
+              'foo');
+          expect(scope.clients.openWindow).toHaveBeenCalledWith('http://localhost:4200/');
+        });
+      });
+
+      describe('No matching onActionClick field', () => {
+        it('opens new window at `/`', async () => {
+          expect(await makeRequest(scope, '/foo.txt')).toEqual('this is foo');
+          spyOn(scope.clients, 'openWindow');
+
+          await driver.initialized;
+          await scope.handleClick(
+              {
+                title: 'This is a test without action',
+                body: 'Test body without action',
+                'data': {
+                  'onActionClick': {
+                    'fooz': {'operation': 'lastFocusedOrOpen', 'url': 'fooz'},
+                  },
+                },
+              },
+              'foo');
+          expect(scope.clients.openWindow).toHaveBeenCalledWith('http://localhost:4200/');
+        });
+      });
+      describe('No action', () => {
+        it('use onActionClick default', async () => {
+          expect(await makeRequest(scope, '/foo.txt')).toEqual('this is foo');
+          spyOn(scope.clients, 'openWindow');
+
+          await driver.initialized;
+          await scope.handleClick({
+            title: 'This is a test without action',
+            body: 'Test body without action',
+            'data': {
+              'onActionClick': {
+                'default': {'operation': 'openWindow', 'url': 'fooz'},
+              },
+            },
+          });
+          expect(scope.clients.openWindow).toHaveBeenCalledWith('http://localhost:4200/fooz');
+        });
+        it('fallback to openWindow at `/`', async () => {
+          expect(await makeRequest(scope, '/foo.txt')).toEqual('this is foo');
+          spyOn(scope.clients, 'openWindow');
+
+          await driver.initialized;
+          await scope.handleClick({
+            title: 'This is a test without action',
+            body: 'Test body without action',
+            'data': {
+              'onActionClick': {},
```

Comment:
```suggestion
            data: {
              onActionClick: {},
```

Although, I think it would be better to omit the `data` property altogether (to ensure everything still works fine).